### PR TITLE
docs: add integration section with docker details

### LINKED ISSUE #2014: Add a docs page for configuring docker for mac.
If you go through any (most?) of the demos on a mac you will have a guaranteedly bad time because your docker vm will not have enough memory because of the fact that our confluent integration requires like 8 Java VMs.

We have [instructions](https://github.com/MaterializeInc/materialize/blob/642c6ff294e8ce52b72d3c4ecf4009e777717917/demo/chbench/README.md#chbench) for what they should be, we should have a consistent place that all our demos and startup instructions can link to.

### PR DESCRIPTION
**Pull request by sploiselle**
@quodlibetor What do you think of something like this as an approach for the Docker docs you mentioned in #2014 ? This seems like it'll be extensible/flexible, e.g.  I was keeping the fact that we might need to start including more details for things like Debezium in mind. However, I don't know if the notion of "integration" is really the right thing.

Fixes #2014

### CODE REVIEW

File: doc/user/integrations/docker.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+Because many of our Docker-based demos leverage a large number of pieces of infrastructure, we recommend running Docker at least **8 GB** of memory.
+
+On macOS:
+
+1. Open **Docker for Mac**'s **Preferences** window.
+1. Click **Resources**.
+1. Click **Advanced**.
+1. Move the **Memory** slider to at least **8.00 GB**.
+1. Click **Apply & Restart**.
```
Comment (benesch) [inline doc/user/integrations/docker.md]: I think this should mention that on Linux Docker will automatically share memory with the host machine, so as long as the host machine has more than 8GB of memory you'll be good to go.

File: doc/user/integrations/docker.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    parent: 'integrations'
+---
+
+Many of our demos rely on Docker and Docker Compose to make it easy to deploy Materialize, along with any other infrastructure the demo needs.
+
+For the best experience using Docker, we recommend following the guidelines outlined here.
+
+### Increase Docker Memory
+
+Because many of our Docker-based demos leverage a large number of pieces of infrastructure, we recommend running Docker at least **8 GB** of memory.
```
Comment (benesch) [inline doc/user/integrations/docker.md]: s/running Docker at least/running Docker with at least/

**Comment by sploiselle** [comment]
@benesch What would you call the "superset" of these guides, including things like Debezium? "External configurations"?

**Comment by quodlibetor** [comment]
> What would you call the "superset" of these guides, including things like Debezium? "External configurations"?

I think I'd call it something like "operating materialize" Debezium is an integration, but the configurations that it requires in postgres are steps that operators take, same thing with changing the docker config.

**Comment by sploiselle** [comment]
Gonna go with Third-Party Tools; we can revisit if there's rampant confusion.

**Pull request merged**